### PR TITLE
GH-242 Added useEncryptedAssertion attribute to SAML SSO Settings Res…

### LINKED ISSUE #242: Missing useEncryptedAssertion attribute from SAML SSO Settings Resource
**Describe the bug**
Looking at the Artifactory GUI or the SAML configuration obtained by using curl from `api/system/configuration` endpoint there is a missing attribute `useEncryptedAssertion` when comparing with [SSO SAML resource](https://registry.terraform.io/providers/jfrog/artifactory/latest/docs/resources/artifactory_saml_settings) that can not be set using this provider.

example from configuration (`api/system/configuration`) one can see that `useEncryptedAssertion` is present.
```
<samlSettings>
    <enableIntegration>true</enableIntegration>
    <verifyAudienceRestriction>false</verifyAudienceRestriction>
    <loginUrl>login url</loginUrl>
    <logoutUrl>log out url</logoutUrl>
    <certificate>your certificate</certificate>
    <serviceProviderName>Provider</serviceProviderName>
    <noAutoUserCreation>true</noAutoUserCreation>
    <allowUserToAccessProfile>true</allowUserToAccessProfile>
    <useEncryptedAssertion>false</useEncryptedAssertion>
    <autoRedirect>true</autoRedirect>
    <syncGroups>true</syncGroups>
    <groupAttribute>groups</groupAttribute>
    <emailAttribute>email</emailAttribute>
</samlSettings>
```

From our production system we know that it is possible to do an http request (patch) to the `artifactory/api/system/configuration` endpoint and configure this attribute/parameter i.e like in the following yaml:
```
  samlSettings:
    enableIntegration: true
    loginUrl: url
    logoutUrl: url
    certificate: value
    serviceProviderName: okta
    noAutoUserCreation: true
    allowUserToAccessProfile: true
    useEncryptedAssertion: false
    autoRedirect: true
    syncGroups: true
    groupAttribute: groups
    emailAttribute: email
```
but for some reason this is not documented anymore here: https://www.jfrog.com/confluence/display/JFROG/Artifactory+YAML+Configuration

saml_mod

Terraform version: v1.0.11 on Ubuntu 20.04
Artifactory provider version(s): 2.6.21 and 2.6.22
Artifactory version: 7.27.6 Revision: 72706900

**Requirements for and issue**
- [x] A fully functioning terraform snippet that can be copy&pasted (no outside files or ENV vars unless that's part of the issue)
- [x] Your version of artifactory (you can `curl` it at `$host/artifactory/api/system/version`
- [x] Your version of terraform

**Expected behavior**

To be able to configure all the possible parameters related to SAML SSO using [Artifactory SAML SSO Settings Resource](https://registry.terraform.io/providers/jfrog/artifactory/latest/docs/resources/artifactory_saml_settings).

### PR DESCRIPTION
**Pull request by maheshjfrog**
…ource
Fixes #242

### CODE REVIEW

File: pkg/artifactory/resource_artifactory_saml_settings.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ type SamlSettings struct {
 	AutoRedirect              bool   `yaml:"autoRedirect" json:"autoRedirect"`
 	SyncGroups                bool   `yaml:"syncGroups" json:"syncGroups"`
 	VerifyAudienceRestriction bool   `yaml:"verifyAudienceRestriction" json:"verifyAudienceRestriction"`
+	UseEncryptedAssertion     bool   `yaml:"useEncryptedAssertion" json:"useEncryptedAssertion"`
```
Comment (alexhung) [inline pkg/artifactory/resource_artifactory_saml_settings.go:34]: Purely my own ignorance. Why is `json` tag here? Does the endpoint `artifactory/api/saml/config` returns json or yaml?

**Review by alexhung — APPROVED** [review]
LGTM

File: pkg/artifactory/resource_artifactory_saml_settings.go (hunk 1 above)
Comment (maheshjfrog) [reply pkg/artifactory/resource_artifactory_saml_settings.go:34]: REST API call uses yaml.

Comment (alexhung) [reply pkg/artifactory/resource_artifactory_saml_settings.go:34]: That's what I assumed. So why do we need the `json` tag?

Comment (maheshjfrog) [reply pkg/artifactory/resource_artifactory_saml_settings.go:34]: Seeing this usage in couple if places. Will test and remove them if all pass in next pass. I have merged this PR already.

Comment (maheshjfrog) [reply pkg/artifactory/resource_artifactory_saml_settings.go:34]: i see ... following for saml resource interactions.
PATCH call to api/system/configuration returns yaml response.
And, GET call to api/saml/config returns json response.

Comment (alexhung) [reply pkg/artifactory/resource_artifactory_saml_settings.go:34]: Ah ok then.

**Pull request merged**